pkg/sloop/common/db_utilities_test.go

```diff
@@ -12,46 +12,29 @@ var commonPrefix = "/commonprefix/001546405200/"
 func Test_Db_Utilities_DeleteKeysWithPrefix_DeleteAllKeys(t *testing.T) {
 	db := helper_get_db(t)
 	helper_add_keys_to_db(t, db, helper_testKeys_with_common_prefix(commonPrefix))
-	err, numOfDeletedKeys, numOfKeysToDelete := DeleteKeysWithPrefix([]byte(commonPrefix), db, 10)
+	err, numOfDeletedKeys, numOfKeysToDelete := DeleteKeysWithPrefix(commonPrefix, db, 10, 4)
 	assert.Nil(t, err)
-	assert.Equal(t, 4, numOfDeletedKeys)
-	assert.Equal(t, 4, numOfKeysToDelete)
+	assert.Equal(t, uint64(4), numOfDeletedKeys)
+	assert.Equal(t, uint64(4), numOfKeysToDelete)
 }
 
 func Test_Db_Utilities_DeleteKeysWithPrefix_DeleteNoKeys(t *testing.T) {
 	db := helper_get_db(t)
 	helper_add_keys_to_db(t, db, helper_testKeys_with_common_prefix(commonPrefix))
-	err, numOfDeletedKeys, numOfKeysToDelete := DeleteKeysWithPrefix([]byte(commonPrefix+"random"), db, 10)
+	err, numOfDeletedKeys, numOfKeysToDelete := DeleteKeysWithPrefix(commonPrefix+"random", db, 10, 0)
 	assert.Nil(t, err)
-	assert.Equal(t, 0, numOfDeletedKeys)
-	assert.Equal(t, 0, numOfKeysToDelete)
+	assert.Equal(t, uint64(0), numOfDeletedKeys)
+	assert.Equal(t, uint64(0), numOfKeysToDelete)
 }
 
 func Test_Db_Utilities_DeleteKeysWithPrefix_DeleteSomeKeys(t *testing.T) {
 	db := helper_get_db(t)
 	helper_add_keys_to_db(t, db, helper_testKeys_with_common_prefix(commonPrefix))
 	helper_add_keys_to_db(t, db, helper_testKeys_with_common_prefix("randomStuff"+commonPrefix))
-	err, numOfDeletedKeys, numOfKeysToDelete := DeleteKeysWithPrefix([]byte(commonPrefix), db, 10)
+	err, numOfDeletedKeys, numOfKeysToDelete := DeleteKeysWithPrefix(commonPrefix, db, 10, 4)
```

Comment:
This test is almost the same as Test_Db_Utilities_DeleteKeysWithPrefix_DeleteAllKeys, maybe we can get count of remain keys to make sure we did not delete all keys.

Reply:
The challenge with delete keys is that its not on a locked db which can lead to a race condition. For example, the keys are getting added for the prefix that you are trying to delete. So in order to avoid it, we return as soon as the number of keys deleted is the desired number. Let me add a comment to add it more clear that in this test case the DB has 8 keys and we are only deleting 4.